extras/beast/unit_test/suite.hpp

```diff
@@ -158,74 +161,103 @@ class suite
     }
 
     /** Invokes the test using the specified runner.
+
         Data members are set up here instead of the constructor as a
         convenience to writing the derived class to avoid repetition of
         forwarded constructor arguments to the base.
         Normally this is called by the framework for you.
     */
     template<class = void>
     void
-    operator() (runner& r);
+    operator()(runner& r);
+
+    /** Record a successful test condition. */
+    template<class = void>
+    void
+    pass();
+
+    /** Record a failure.
+
+        @param reason 
+    */
+    template<class = void>
+    void
+    fail(std::string const& reason = "");
 
     /** Evaluate a test condition.
-        The condition is passed as a template argument instead of `bool` so
-        that implicit conversion is not required. The `reason` argument is
-        logged if the condition is false.
+
+        This function provides improved logging by incorporating the
+        file name and line number into the reported output on failure,
+        as well as additional text specified by the caller.
+
+        @param shouldBeTrue The condition to test. The condition
+        is evaluated in a boolean context.
+
+        @param reason Optional added text to output on a failure.
+
+        @param file The source code file where the test failed.
+
+        @param line The source code line number where the test failed.
+
         @return `true` if the test condition indicates success.
     */
-    template<class Condition, class String>
+    /** @{ */
+    template<class Condition>
     bool
-    expect(Condition const& shouldBeTrue,
-        String const& reason);
+    expect(Condition const& shouldBeTrue)
+    {
+        return expect(shouldBeTrue, {});
+    }
 
     template<class Condition>
     bool
-    expect(Condition const& shouldBeTrue)
+    expect(Condition const& shouldBeTrue, std::string const& reason);
+
+    template<class Condition>
+    bool
+    expect(Condition const& shouldBeTrue,
+        char const* file, int line)
     {
-        return expect(shouldBeTrue, "");
+        return expect(shouldBeTrue, {}, file, line);
     }
 
-    /** Expect an exception from f() */
-    /** @{ */
-    template<class F, class String>
+    template<class Condition>
     bool
-    except (F&& f, String const& reason);
+    expect(Condition const& shouldBeTrue,
+        std::string const& reason, char const* file, int line);
+    /** @} */
 
+    //
+    // DEPRECATED
```

Comment:
If this was an official API then yeah I would consider it but its in the extras/ directory which means its not official (but still available for use if you accept the risks).